front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -100,16 +118,41 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
-  const [tmpSelectedTrain, setTmpSelectedTrain] = useState(selectedTrainScheduleId);
-
-  useEffect(() => {
-    setTmpSelectedTrain(selectedTrainScheduleId);
-  }, [selectedTrainScheduleId]);
+  const projectedTrainsAndOccurrences = useMemo(
+    () =>
+      projectPathTrainResult.flatMap((train) => {
+        if (isOccurrence(train.id) && train.paced) {
+          const pacedTrainId = getPacedTrainIdFromOccurrenceId(train.id);
+          const occurrencesCount = Math.ceil(
+            Duration.parse(train.paced.duration).ms / Duration.parse(train.paced.step).ms
+          );
+          const occurrences = [];
+          for (let i = 0; i < occurrencesCount; i += 1) {
+            const occurrenceStartTime = dayjs(train.departureTime)
+              .add(i * Duration.parse(train.paced.step).ms, 'ms')
+              .toDate();
+            occurrences.push({
+              id: `occurrence-${i}-${pacedTrainId}` as OccurrenceId,
+              name: computeOccurrenceName(train.name, i),
+              departureTime: occurrenceStartTime,
+            });
```

Comment:
You should push directly the full occurence
```suggestion
            occurrences.push({
              ...train,
              name: computeOccurrenceName(train.name, i),
              departureTime: occurrenceStartTime,
            });
```

Reply:
i must generate an id but to find a resolution in between i can effectively go for 
 ```js
occurrences.push({
              ...train,
              id: `occurrence-${i}-${pacedTrainId}` as OccurrenceId,
              name: computeOccurrenceName(train.name, i),
              departureTime: occurrenceStartTime,
            });
          }
return occurrences;
```

Reply:
The id is already in the train no ?

Reply:
The id we have in train is that of the 1st occurrence, we need to recreate different ids for each occurrence

Reply:
My bad, I thought we already had the occurence id. Yes I'm ok with what you suggested @Math-R, you should just use the function to create the occurenceId rather than create it here with a cast

Reply:
done